File: go-ds-and-algorithms/sort/selection_sort.go
package sort

func selectionSort(nums []int) {
	l := len(nums)
	if l <= 1 {
		return
	}

	for i := 0; i < l; i++ {
		min := i
		for j := i + 1; j < l; j++ {
			if nums[min] > nums[j] {
				min = j
			}
		}
		// 如果 min 就是最小元素，就不交换了
		if min != i {
			nums[i], nums[min] = nums[min], nums[i]
		}
	}
}

File: java-ds-and-algorithms/src/main/java/sort/SelectionSort.java
package sort;

/**
 * Usage: 选择排序 <br/>
 * Date: 2023/5/3 22:05 <br/>
 *
 * @author <a href="mailto:hanzhang2566@foxmail.com">hanzhang</a>
 */
public class SelectionSort {
    public void sort(int[] nums) {
        int length = nums.length;
        if (length <= 1) {
            return;
        }

        for (int i = 0; i < length; i++) {
            int min = i;
            for (int j = i + 1; j < length; j++) {
                if (nums[min] > nums[j]) {
                    min = j;
                }
            }
            if (min != i) {
                swap(nums, min, i);
            }
        }
    }

    private void swap(int[] nums, int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}
